packages/fast-jss-manager-react/src/manage-jss.tsx

```diff
@@ -131,13 +123,43 @@ function manageJss<S, C>(styles?: ComponentStyles<S, C>): <T>(Component: React.C
                 return this.context && this.context.designSystem ? this.context.designSystem : {};
             }
 
+            /**
+             * Remove a JSS stylesheet
+             */
+            private removeStyleSheet(): void {
+                if (this.hasStyleSheet()) {
+                    this.state.styleSheet.detach();
+                    jss.removeStyleSheet(this.state.styleSheet);
+                }
+            }
+
+            /**
+             * Reset a JSS stylesheet relative to current props
+             */
+            private resetStyleSheet(): any {
+                this.removeStyleSheet();
+                this.setState((previousState: IJSSManagerState, props: T & IJSSManagerProps<S, C>): Partial<IJSSManagerState> => {
+                    return {
+                        styleSheet: this.hasStyleSheet() ? this.createStyleSheet() : null
+                    };
+                }, (): void => {
+                    if (this.hasStyleSheet()) {
+                        this.state.styleSheet.attach().update(this.designSystem);
+                    }
+                });
+            }
+
             /**
              * Creates a jss stylesheet from the dynamic portion of an associated style object and any style object passed
```

Comment:
nit: JSS is referenced should be the same casing.  So either jss or JSS.  I'd suggest JSS since it's an acronym.  Though in title comments i'd use "feat(jss)..." for consistency and cleaner change logs.  Just my thoughts